commands/social/store/store.js

```diff
@@ -19,70 +20,33 @@ class Store extends Social {
     }
     // console.log(message.guild.store);
     switch (message.flags[0]) {
-      case ("buy"): {
+      case "buy": {
         const name = args.join(" ");
-        
-        if (!name) return this.client.commands.get("store").run(message, args, level);
-            
-        const item = this.client.store.filter(i => i.name.toLowerCase().includes(name.toLowerCase()));
-        
-        if (item.size > 1) return message.reply(`B..Baka! Be more specific more than that, there is more than one item on sale with ${name} as their name`);
-        if (!item) return message.channel.send("That item doesn't exist, Please make sure it is spelled correctly");
-        if (message.member.roles.has(item.array()[0].id)) return message.channel.send("You already have the role :facepalm: ");
-        
-        if (parseInt(item.array()[0].price) > parseInt(message.member.score.points)) {
-          return message.channel.send(`You currently have ₲${parseInt(message.member.score.points).toLocaleString()}, but the role costs ${parseInt(item.array()[0].price).toLocaleString()}!`);
-        }
-
-        const filter = m => m.author.id === message.author.id;
-        const response = await this.client.awaitReply(message, `Are you sure you want to purchase ${item.array()[0].name} for ₲${item.array()[0].price.toLocaleString()}?`, filter, undefined, null);
-        if (["y", "yes"].includes(response.toLowerCase())) {
-        
-          message.member.takePoints(parseInt(item.array()[0].price));
-          await message.member.roles.add(item.array()[0].id);
-          message.channel.send("You have bought the role :tada: ");
-        
-        } else
-        
-        if (["n", "no", "cancel"].includes(response.toLowerCase())) {
-          message.response(undefined, "Transaction cancelled.");
+        if (!name) return message.reply(`|\`❌\`| ${this.help.usage}`);
+        if (name.toLowerCase() === "slots") {
+          this.buyToken(name, message, this.client);
```

Comment:
please await all methods which return a Promise in here this should be done in the whole PR